lib/everest.cpp

```diff
@@ -801,6 +812,8 @@ void Everest::handle_ready(const json& data) {
         on_ready_handler();
     }
 
+    this->ready_received = true;
```

Comment:
I'd argue "ready received" happens as soon as the message is received, this would be more of a "ready handler called". maybe we need 2 vars to track this here